site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -20,6 +20,12 @@
  * @AccessControl(role="INSTRUCTOR")
  */
 class AdminGradeableController extends AbstractController {
+    /**
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/homework_library", methods={"GET"})
+     */
+    private function selectFromHomeworkLibrary($gradeable_id, $semester, $course) {
+        $this->core->getOutput()->renderTwigOutput('admin/admin_gradeable/AdminGradeableHomeworkLibrary.twig', []);
```

Comment:
HomeworkLibrary.twig and admin/admin_gradeable/AdminGradeableHomeworkLibrary.twig are the same. Could this use HomeworkLibrary.twig? Don't want two different search pages.

Reply:
Carman told me to do it that way.

Reply:
Ok yeah sounds like the HomeworkLibrary.twig should be removed and HomeworkView should use this one instead.

Reply:
What is the error message you're getting @JarodO ? Might it have anything to do with the fact that you aren't using any of the paramaters you've created in AdminGradeableController ($gradeable_id, $semester, $course)?

Reply:
@N-Perry Here it is: [Link](https://pastebin.com/RQnsKj2A)

Reply:
Oh! I just remembered, you should use their Response class to generate the response. I think this is their old way of doing things. But, if you must, I suppose they still use it and it SHOULD work even though it's not pretty...